ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java

```diff
@@ -172,154 +156,56 @@ public boolean containsTable(ServerTableIdentifier identifier) {
-    authOptimizers.entrySet().removeIf(op -> op.getValue().getResourceId().equals(resourceId));
-  }
-
-  private void clearTasks(TableOptimizingProcess optimizingProcess) {
-    retryQueue.removeIf(
+  private void clearProcess(TableOptimizingProcess optimizingProcess) {
+    tableQueue.removeIf(process -> process.getProcessId() == optimizingProcess.getProcessId());
+    retryTaskQueue.removeIf(
         taskRuntime -> taskRuntime.getProcessId() == optimizingProcess.getProcessId());
-    taskQueue.removeIf(
-        taskRuntime -> taskRuntime.getProcessId() == optimizingProcess.getProcessId());
-    executingTaskMap
-        .entrySet()
-        .removeIf(entry -> entry.getValue().getProcessId() == optimizingProcess.getProcessId());
-  }
-
-  @Override
-  public void ping() {}
-
-  @Override
-  public void touch(String authToken) {
-    OptimizerInstance optimizer = getAuthenticatedOptimizer(authToken).touch();
-    LOG.debug("Optimizer {} touch time: {}", optimizer.getToken(), optimizer.getTouchTime());
-    doAs(OptimizerMapper.class, mapper -> mapper.updateTouchTime(optimizer.getToken()));
-  }
-
-  private OptimizerInstance getAuthenticatedOptimizer(String authToken) {
-    Preconditions.checkArgument(authToken != null, "authToken can not be null");
-    return Optional.ofNullable(authOptimizers.get(authToken))
-        .orElseThrow(() -> new PluginRetryAuthException("Optimizer has not been authenticated"));
   }
 
-  @Override
-  public OptimizingTask pollTask(String authToken, int threadId) {
-    getAuthenticatedOptimizer(authToken);
-    TaskRuntime task = Optional.ofNullable(retryQueue.poll()).orElseGet(this::pollOrPlan);
-
-    if (task != null) {
-      safelySchedule(task, new OptimizingThread(authToken, threadId));
-      executingTaskMap.putIfAbsent(task.getTaskId(), task);
+  public TaskRuntime pollTask(long maxWaitTime) {
+    long deadline = calculateDeadline(maxWaitTime);
+    TaskRuntime task = fetchTask();
+    while (task == null && waitTask(deadline)) {
+      task = fetchTask();
     }
-    return task != null ? task.getOptimizingTask() : null;
+    return task;
   }
 
-  private void safelySchedule(TaskRuntime task, OptimizingThread thread) {
-    try {
-      task.schedule(thread);
-    } catch (Throwable throwable) {
-      LOG.error("Schedule task {} failed, put it to retry queue", task.getTaskId(), throwable);
-      retryTask(task, false);
-      throw throwable;
-    }
+  private long calculateDeadline(long maxWaitTime) {
+    long deadline = System.currentTimeMillis() + maxWaitTime;
+    return deadline <= 0 ? Long.MAX_VALUE : deadline;
   }
 
-  private void retryTask(TaskRuntime taskRuntime, boolean incRetryCount) {
-    taskRuntime.reset(incRetryCount);
-    retryQueue.offer(taskRuntime);
+  private TaskRuntime fetchTask() {
+    return Optional.ofNullable(retryTaskQueue.poll())
+        .orElse(
+            Optional.ofNullable(tableQueue.peek()).map(TableOptimizingProcess::poll).orElse(null));
```

Comment:
As long as the TaskQueue is not empty, the plan will not be triggered. It will only be triggered when the TaskQueue is empty. Even when the TaskQueue is empty, it will still wait for a table plan to be completed, making it difficult to demonstrate the effects of multithreading. Perhaps our plan thread's triggering strategy can be based on the capacity of the TaskQueue, automatically triggering the plan in the background when the TaskQueue is relatively empty.